subsys/bluetooth/audio/pacs.c

```diff
@@ -271,30 +262,31 @@ static int set_supported_contexts(uint16_t contexts, uint16_t *supported,
 		return 0;
 	}
 
-	err = k_work_reschedule(&supported_contexts_work, PAC_NOTIFY_TIMEOUT);
-	if (err < 0) {
-		return err;
-	}
-
 	*supported = contexts;
 
 	/* Update available contexts if needed*/
 	if ((contexts & *available) != *available) {
-		*available = *available & contexts;
-		err = k_work_reschedule(&available_contexts_work,
-					PAC_NOTIFY_TIMEOUT);
-		if (err < 0) {
-			LOG_WRN("Update available contexts notify failed: %d", err);
+		err = set_available_contexts(contexts & *available, available, contexts);
+		if (err) {
+			*available = tmp_available;
+
+			return err;
```

Comment:
At this point `*supported` has also been updated, so that needs to be restored as well